### ISSUE TITLE
Could Captum be used for sequence tagging problems?

### ISSUE THREAD

**Issue by heytitle** [pytorch/captum#374]
Hi,

I'm currently working on a sequence tagging problem, and I'm interested in investigating what tokens in the sequence are relevant for each step's predicting.

**Comment by NarineK** [comment]
Hi @heytitle , you should be able to use captum for sequence tagging. We don't have a tutorial on that but I'd recommend to start from IG and attribute predicted output at each step to the input embeddings. Here is an older example on translation. https://github.com/pytorch/captum/blob/4378c1c7ae733a56fa781881faa3cf59b8829b6b/tutorials/seq2seq_interpret.ipynb

**Comment by heytitle** [comment]
@NarineK Thanks for the reply!

Just out of my curiosity, what don't we have that tutorial on the website?

**Comment by NarineK** [comment]
It was long time ago. I didn't get time to review it. Seq2seq models can also be more computationally expensive but we can definitely think of adding a tutorial. Do you have any recommendations on seq2seq models in PyTorch that would be good to have a tutorial on?

**Comment by heytitle** [comment]
I'll take a look and let you know if I come up with something. The problem that I'm currently working on right now is about word segmentation for [Thai language](https://arxiv.org/abs/1911.07056). It's quite specific and might not be a good example to illustrate Captum's features.

**Comment by LopezGG** [comment]
@NarineK : I used your tutorial to create one for seq2seq model (i.e NER) [here](https://github.com/LopezGG/IntrepretingSeq2Seq/blob/master/NER_Model_Interpret.ipynb) . Could you review it ?

**Comment by lipka-clazzpl** [comment]
Hi @LopezGG, great tutorial, 

>We also find that the last few layers of the target word "Rob" also has a strong positive influence.

Actually last few layers (starting with 1 up to layer 4) are quite cold, think You wanted to point out to layers 5-7.

**Comment by LopezGG** [comment]
@lipka-clazzpl : Thanks for your feedback. I believe the lighter color refers to positive feedback (0.8-1.0) based on the scale here. Am I missing something
![image](https://user-images.githubusercontent.com/1724189/89435123-491c1780-d6f9-11ea-8e76-e89e1a5efd11.png)

**Comment by lipka-clazzpl** [comment]
@LopezGG correct, lighter color=more positive feedback, thought You wanted to explain attribution for word "Rob" solely with all the layers, my misinterpretation, You clearly explain "Rob" with "win", ",", "in" and "surprise" at lowest layers.

**Comment by craymichael** [comment]
This issue appears to have been answered in the discussion above, so I will close it for now. Please feel free to reopen or follow up if anything is still unresolved.
